### PR TITLE
Add a bindable Enabled flag to NotificationManager

### PR DESCRIPTION
**Pull request by peppy**
Also better handles delays before notifications are displayed. After being re-enabled, the overlay will wait 1 second before starting to post notifications to the user.

### CODE REVIEW

File: osu.Game/Overlays/NotificationOverlay.cs [hunk 1]
```diff
@@ -84,9 +112,13 @@ private void notificationClosed()
                 State = Visibility.Hidden;
         }
 
+        private readonly Scheduler postScheduler = new Scheduler();
+
+        private bool processingPosts = true;
```
Comment (smoogipoo) [inline osu.Game/Overlays/NotificationOverlay.cs]: Can just use `postScheduler.Enabled` instead of local bool.

File: osu.Game/OsuGame.cs [hunk 2]
```diff
@@ -357,6 +374,8 @@ private void hideAllOverlays()
             notifications.State = Visibility.Hidden;
         }
 
+        private ScheduledDelegate notificationsEnabler;
```
Comment (smoogipoo) [inline osu.Game/OsuGame.cs]: Unused?

**Review by smoogipoo — CHANGES_REQUESTED** [review]

File: osu.Game/Overlays/NotificationOverlay.cs (hunk 1 above)
Comment (peppy) [reply osu.Game/Overlays/NotificationOverlay.cs]: used to add the delay (without affecting the `BindableBool`'s state

File: osu.Game/OsuGame.cs (hunk 2 above)
Comment (peppy) [reply osu.Game/OsuGame.cs]: yeah, good catch

File: osu.Game/Overlays/NotificationOverlay.cs (hunk 1 above)
Comment (smoogipoo) [reply osu.Game/Overlays/NotificationOverlay.cs]: I mean the Scheduler's Enabled bool, not the bindable.

Comment (peppy) [reply osu.Game/Overlays/NotificationOverlay.cs]: only available in `ThreadedScheduler` :(

**Comment by smoogipoo** [comment]
What's the verdict here? Do we still need the notification part of this?

**Comment by peppy** [comment]
Yeah i want it for now

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**